ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -940,31 +1129,130 @@ const NetworksForm = ({
           />
         ) : (
           <FormField
-            error={errors.rpcUrl?.msg || ''}
             onChange={(value) => {
               setIsEditing(true);
               setRpcUrl(value);
             }}
             titleText={t('rpcUrl')}
             value={displayRpcUrl}
-            disabled={viewOnly}
+            disabled={
+              (viewOnly ||
+                isDefaultMainnet ||
+                isDefaultLineaMainnet ||
+                isDefaultLineaSepoliaTestnet ||
+                isDefaultSepoliaTestnet) &&
+              !addNewNetwork
+            }
             dataTestId="network-form-rpc-url"
           />
         )}
+        {errors.rpcUrl?.msg ? (
+          <HelpText
+            severity={HelpTextSeverity.Danger}
+            marginTop={1}
+            data-testid="network-form-rpc-url-error"
+          >
+            {errors.rpcUrl.msg}
+          </HelpText>
+        ) : null}
+
         <FormField
-          warning={warnings.chainId?.msg || ''}
-          error={errors.chainId?.msg || ''}
           onChange={(value) => {
             setIsEditing(true);
             setChainId(value);
             autoSuggestTicker(value);
+            autoSuggestName(value);
           }}
           titleText={t('chainId')}
           value={chainId}
-          disabled={viewOnly}
+          disabled={
+            (viewOnly ||
+              isDefaultMainnet ||
+              isDefaultLineaMainnet ||
+              isDefaultLineaSepoliaTestnet ||
+              isDefaultSepoliaTestnet ||
+              isPopularNetwork) &&
+            !addNewNetwork
```

Comment:
Since this are so many conditions, I think we should put this into a `const` above so it's separated from the JSX.